src/Job.php

```diff
@@ -62,14 +62,14 @@ public function delete()
      */
     public function release($delay = 0)
     {
-        if ($delay) {
-            throw new \LogicException('To be implemented');
-        }
+        parent::release($job);
 
         $requeueMessage = clone $this->psrMessage;
         $requeueMessage->setProperty('x-attempts', $this->attempts() + 1);
 
-        $this->psrContext->createProducer()->send($this->psrConsumer->getQueue(), $requeueMessage);
+        $this->psrContext->createProducer()
+                ->setDeliveryDelay($this->secondsUntil($delay))
```

Comment:
the setter can throw an exception in case delaying is not supported. I think we should `$this->psrConsumer->acknowledge($this->psrMessage);` before trying to requeue.

Reply:
```
the setter can throw an exception in case delaying is not supported.
```
i think we need an api for that, is it alright to have something like

```
trait DelayStrategyAwareTrait
{
    // ..............

    public function delayable()
    {
        return $this->delayStrategy ? true : false;
    }
}
```

also my pr code should check for ```DelayStrategyAware``` contract`


```
I think we should $this->psrConsumer->acknowledge($this->psrMessage); before trying to requeue
```
yes i agree